packages/terra-tabs/src/_CollapsibleTabs.jsx

```diff
@@ -73,10 +73,13 @@ class CollapsibleTabs extends React.Component {
     this.handleSelectionAnimation();
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps) {
     if (this.isCalculating) {
       this.isCalculating = false;
       this.handleResize(this.contentWidth);
+    } else if (React.Children.count(this.props.children) !== React.Children.count(prevProps.children)) {
```

Comment:
One thing to note is that I don't think this will resolve issues with tab reordering or tab swaps (removing 1 tab and adding 1 tab at the same time) because the count will remain the same, but until a user requests this functionality I would not make code changes for it.